meta/serialize.pm

```diff
@@ -847,18 +848,18 @@ sub CreateSerializeEmitMacros
 {
     WriteSectionComment "Emit macros";
 
-    WriteSource "#define EMIT(x)        buf += sprintf(buf, x)";
-    WriteSource "#define EMIT_QUOTE     EMIT(\"\\\"\")";
-    WriteSource "#define EMIT_KEY(k)    EMIT(\"\\\"\" k \"\\\":\")";
-    WriteSource "#define EMIT_NEXT_KEY(k) { EMIT(\",\"); EMIT_KEY(k); }";
-    WriteSource "#define EMIT_CHECK(expr, suffix) {                                 \\";
-    WriteSource "    ret = (expr);                                                  \\";
-    WriteSource "    if (ret < 0) {                                                 \\";
-    WriteSource "        SAI_META_LOG_WARN(\"failed to serialize \" #suffix \"\");      \\";
-    WriteSource "        return SAI_SERIALIZE_ERROR; }                              \\";
-    WriteSource "    buf += ret; }";
-    WriteSource "#define EMIT_QUOTE_CHECK(expr, suffix) {\\";
-    WriteSource "    EMIT_QUOTE; EMIT_CHECK(expr, suffix); EMIT_QUOTE; }";
+    WriteHeader "#define EMIT(x)        buf += sprintf(buf, x)";
+    WriteHeader "#define EMIT_QUOTE     EMIT(\"\\\"\")";
+    WriteHeader "#define EMIT_KEY(k)    EMIT(\"\\\"\" k \"\\\":\")";
+    WriteHeader "#define EMIT_NEXT_KEY(k) { EMIT(\",\"); EMIT_KEY(k); }";
+    WriteHeader "#define EMIT_CHECK(expr, suffix) {                                 \\";
+    WriteHeader "    ret = (expr);                                                  \\";
+    WriteHeader "    if (ret < 0) {                                                 \\";
+    WriteHeader "        SAI_META_LOG_WARN(\"failed to serialize \" #suffix \"\");      \\";
+    WriteHeader "        return SAI_SERIALIZE_ERROR; }                              \\";
+    WriteHeader "    buf += ret; }";
+    WriteHeader "#define EMIT_QUOTE_CHECK(expr, suffix) {\\";
+    WriteHeader "    EMIT_QUOTE; EMIT_CHECK(expr, suffix); EMIT_QUOTE; }";
```

Comment:
move this back to source, i specially put that in source file to not collide with other defines, this cant be in header

Reply:
sure, im fine with duplicate

Reply:
ACK